In [10]:
from pathlib import Path
import os
import pandas as pd
from helpers.utils import dice_score
from helpers import paths
from helpers.paths import (
    RESOURCES_DIR, TRAIN_ROOT, PROJECT_ROOT
)
from helpers.shell_interface import open_itksnap_workspace_cmd

In [4]:
subject_sessions = pd.read_csv(RESOURCES_DIR/"subject-sessions.csv", index_col="sub")
subjects_file = PROJECT_ROOT / "training/roi_train2/subjects.txt"
with open(subjects_file, 'r') as f:
    subjects = [line.strip() for line in f.readlines()]

In [6]:
from collections import defaultdict
dataroot = Path("/media/smbshare/3Tpioneer_bids")
label_names = [
    "lesion.t3m20/prl_mask_def_prob_LR.nii.gz", 
    "lesion.t3m20/prl_mask_def_prob_CH.nii.gz",
    "lesion.t3m20/prl_mask_def_prob_SRS.nii.gz",
    "lesion.t3m20/prl_mask_def_prob_SRS_CH.nii.gz",
]

labels_to_check = defaultdict(list)
for subid in subjects:
    sesid = subject_sessions.loc[int(subid), "ses"]
    subject_root = dataroot / f"sub-ms{subid}" / f"ses-{sesid}"
    for n in label_names:
        p = subject_root / n
        if p.exists():
            labels_to_check[subid].append(p)

In [7]:
for k,v in labels_to_check.copy().items():
    if len(v) < 2:
        labels_to_check.pop(k)

In [13]:
labels_to_check.keys()

dict_keys(['1011', '1038', '1074', '1080', '1215', '1293', '1348', '1396', '2026', '2041'])

In [12]:
import subprocess
subid = "1011"
labels = labels_to_check['1011']
subject_root = labels[0].parent.parent
images = [subject_root / im for im in ["flair.nii.gz", "phase.nii.gz"]]
for lab in labels:
    print(lab.name)
cmd = open_itksnap_workspace_cmd(images, labels)
subprocess.run(cmd, shell=True)

prl_mask_def_prob_LR.nii.gz
prl_mask_def_prob_SRS.nii.gz
prl_mask_def_prob_SRS_CH.nii.gz
Launching ITK-SNAP


Loaded translator for locale "en_US"
Loaded qtbase translator for locale "en_US"
OpenGL Vendor: NVIDIA Corporation
OpenGL Renderer: NVIDIA GeForce RTX 4090/PCIe/SSE2
OpenGL Version: 3.2.0 NVIDIA 580.126.20
OpenGL Vendor: NVIDIA Corporation
OpenGL Renderer: NVIDIA GeForce RTX 4090/PCIe/SSE2
OpenGL Version: 3.2.0 NVIDIA 580.126.20
OpenGL Vendor: NVIDIA Corporation
OpenGL Renderer: NVIDIA GeForce RTX 4090/PCIe/SSE2
OpenGL Version: 3.2.0 NVIDIA 580.126.20
QPainter Render Timing: FBO = 31.821 ms, Blit = 3.065 ms
Return code : 0


CompletedProcess(args='itksnap -g /media/smbshare/3Tpioneer_bids/sub-ms1011/ses-20180911/flair.nii.gz -o /media/smbshare/3Tpioneer_bids/sub-ms1011/ses-20180911/phase.nii.gz -s /media/smbshare/3Tpioneer_bids/sub-ms1011/ses-20180911/lesion.t3m20/prl_mask_def_prob_LR.nii.gz /media/smbshare/3Tpioneer_bids/sub-ms1011/ses-20180911/lesion.t3m20/prl_mask_def_prob_SRS.nii.gz /media/smbshare/3Tpioneer_bids/sub-ms1011/ses-20180911/lesion.t3m20/prl_mask_def_prob_SRS_CH.nii.gz', returncode=0)

1011: LR is not subsegmented
1038: CH and LR; 2 PRL
1074: SRS is not subsegmented
1080: SRS and LR; 2 PRL
1215: SRS and CH

In [1]:
labels_to_check = {"images": [], "labels": []}
labels_to_check['images'].append(1)
labels_to_check

{'images': [1], 'labels': []}